src/lager_app.erl

```diff
@@ -151,23 +158,24 @@ start_error_logger_handler(_, HWM, {ok, WhiteList}) ->
                         throw({error, bad_config})
                 end,
 
-    case supervisor:start_child(lager_handler_watcher_sup, [error_logger, error_logger_lager_h, [HWM, GlStrategy]]) of
+
+    _ = case supervisor:start_child(lager_handler_watcher_sup, [error_logger, error_logger_lager_h, [HWM, GlStrategy]]) of
         {ok, _} ->
             [begin error_logger:delete_report_handler(X), X end ||
                 X <- gen_event:which_handlers(error_logger) -- [error_logger_lager_h | WhiteList]];
         {error, _} ->
             []
-    end.
+    end,
 
-%% `determine_async_behavior/3' is called with the results from either
-%% `application:get_env/2' and `proplists:get_value/2'. Since
-%% `application:get_env/2' wraps a successful retrieval in an `{ok,
-%% Value}' tuple, do the same for the result from
-%% `proplists:get_value/2'.
-wrap_proplist_value(undefined) ->
-    undefined;
-wrap_proplist_value(Value) ->
-    {ok, Value}.
+    Handlers = case application:get_env(lager, handlers) of
```

Comment:
Decided to leave this alone because it's clear enough what is going on here and there are other places where `application:gen_env/2` is called and I guess if I'm going to be pedantic about this, I might as well be pedantic about it everywhere in the application.